src/js/components/DataTable/stories/OnUpdate.js

```diff
@@ -7,89 +8,219 @@ import { grommet } from 'grommet/themes';
+    const result = a.id.localeCompare(b.id, 'en', { sensitivity: 'base' });
     return dir === 'desc' ? -result : result;
   });
 
-  groups.forEach(id => {
-    const data = allData.filter( item => item.location === id)
-      .map(item => ({ id: item.name, ...item }));
-    const paid = data.reduce((prev, curr) => prev + curr.paid, 0);
-    if (data.length) {
+  // add any non-empty groups to items. Also add members of
+  // expanded groups.
+  groups.forEach(({id, members}) => {
+    if (members.length > 0) {
       if (id) {
-        items.push({ id, location: id, paid });
+        const paid = members.reduce((prev, curr) => prev + curr.paid, 0);
+        items.push({id, name: id, paid});
       }
+    
       if (!id || expanded?.includes(id)) {
         if (sort?.property) {
-          data.sort((a,b) => {
-            const v1 = a[sort.property] || '';
-            const v2 = b[sort.property] || '';
-            const dir = sort.direction || 'asc';
-            let result = 0;
-            if (typeof v1 === 'string' && typeof v2 === 'string') {
-              result = v1.localeCompare(v2, 'en', { sensitivity: 'base' });
-            }
-            else if (v1 === v2) {
-              result = 0;
-            } else if (v1 < v2) {
-              result = -1;
-            }
-            else {
-              result = 1;
-            }
-            return dir === 'desc' ? -result : result;
-          });
+          members.sort((a,b) => sortCompare(a, b, sort));
         }
-        items.push(...data);
+        items.push(...members);
       }
     }
   });
-  // TODO support paginate 
-  let start = show || 0;
-  if (start + count > items.length) {
-    start = Math.max(items.length - count, 0);
-  }
-  const result = count ? items.slice(start, start + count) : items;
-  console.log('getData', groups, expanded, sort, show, count, result);
+
+  const start = show || 0;
+ 
+  const result = count ? items.slice(0, Math.max(count, start + count)) : items;
   return result;
 };
 
 export const OnUpdateDataTable = () => {
   const step = 5;
-  const [expand, setExpand] = useState(['Boise'] /* expandable */);
-  const [data, setData] = useState(() => 
-    getData({
+  const [groupSelected, setGroupSelected] = useState({});
+  const [select, setSelect] = useState([]);
+  const [expand, setExpand] = useState(['Fort Collins']);
+  const [data, setData] = useState(() => getData({
       expanded: expand,
       sort: { property: 'name', direction: 'asc'},
       count: step,
       // show: 2,
-    }));
+    }),
+  );
+  
+  const onSelect = useCallback((selected, row) => {
+    const groupUpdates = { };
+    if (row?.location) {
+      // this is a member of a group. Update the group selection state
+      const memberKeys = groupMap[row.location].members.map(({id}) => id);
+      const selectedMembers = selected.filter((s) => memberKeys.includes(s));
+      if (selectedMembers.length === 0) {
+        groupUpdates[row.location] = 'none';
```

Comment:
Since we are referencing `none`, `some`, etc. often as strings, would it be good to extract this as a constant:
```
const selected = {
   none: 'none', 
   some: 'some',
   all: 'all'
};
```
Then reference as `selected .none`?

Not necessary but just to avoid string typos